src/messages/mod.rs

```diff
@@ -152,6 +155,7 @@ impl Message {
             signature_share,
         };
         let src = SrcAuthority::BlsShare {
+            src_section,
```

Comment:
Are we ever going to use anything else than `name(&public_key)` (that is, the name of the sending node) here?

Reply:
We now only send in `self.section().prefix().name()` to the method where this is constructed, and I suppose that is the only thing we want there, as passing in a node name would(?) give different behaviour if a response were to be sent after the src section splits. Actually, if sending to a parent name, would both children be equally close? Is the msg sent to both the children sections then?

Reply:
If you set it to `prefix.name()` and the section splits, then the response will be received only by one of the subsections (the one last prefix bit is 0). Maybe that's OK?

> Actually, if sending to a parent name, would both children be equally close? Is the msg sent to both the children sections then?

No, would still be sent only to one subsection. The sections are disjoint which means there is always only one section that matches any given name. So there is no way to send to multiple sections while giving the destination as a name only.

Reply:
Maybe best to handle this on a case-by-case basis. So for example, if a section is relaying a message from a client, then this should perhaps be set to the client's name. So the response goes to whichever section is still responsible for the client?